cupy/lib/polynomial.py

```diff
@@ -0,0 +1,286 @@
+import re
+import warnings
+
+import cupy
+from cupy.core.overrides import set_module
+
+
+@set_module('cupy')
+class RankWarning(UserWarning):
+    """Issued when the Vandermonde matrix is rank deficient.
+
+    .. seealso:: :func:`numpy.RankWarning`
+
+    """
+    pass
+
+
+_poly_mat = re.compile(r'[*][*]([0-9]*)')
+
+
+def _raise_power(astr, wrap=70):
+    n = 0
+    line1 = ''
+    line2 = ''
+    output = ' '
+    while True:
+        mat = _poly_mat.search(astr, n)
+        if mat is None:
+            break
+        span = mat.span()
+        power = mat.groups()[0]
+        partstr = astr[n:span[0]]
+        n = span[1]
+        toadd2 = partstr + ' ' * (len(power) - 1)
+        toadd1 = ' ' * (len(partstr) - 1) + power
+        if ((len(line2) + len(toadd2) > wrap) or
+                (len(line1) + len(toadd1) > wrap)):
+            output += line1 + '\n' + line2 + '\n '
+            line1 = toadd1
+            line2 = toadd2
+        else:
+            line2 += partstr + ' ' * (len(power) - 1)
+            line1 += ' ' * (len(partstr) - 1) + power
+    output += line1 + '\n' + line2
+    return output + astr[n:]
+
+
+@set_module('cupy')
```

Comment:
Conventionally we don't override modules. For example, we don't override `cupy.core.core.ndarray`.

Reply:
I think it's necessary for NumPy compatibility, see #3266. But it should be done in a separate PR.